lib/xclarity_client/xclarity_base.rb

```diff
@@ -16,7 +16,11 @@ def initialize(conf, uri)
     def connection_builder(conf, uri)
       $lxca_log.info "XClarityClient::XClarityBase connection_builder", "Creating connection to #{conf.host + uri}" 
       #Building configuration
-      @conn = Faraday.new(url: conf.host + uri) do |faraday|
+      host = conf.host;
+      if !host.start_with?('https://') && !host.start_with?('http://')
+         host = 'https://' + host
```

Comment:
The other option is to push some of this logic up into the provider and have the provider add the protocol to be used if it is not specified.